RocketChat/Rocket.Chat
# [FIX] Prevent users to edit an existing role when adding a new one with the same name used before.

## Proposed changes (including videos or screenshots)


### before
![Peek 2021-07-13 16-31](https://user-images.githubusercontent.com/27704687/125513721-953d84f4-1c95-45ca-80e1-b00992b874f6.gif)

### after
![Peek 2021-07-13 16-34](https://user-images.githubusercontent.com/27704687/125514098-91ee8014-51e5-4c62-9027-5538acf57d08.gif)




## Issue(s)


## Steps to test or reproduce

- Go to admin and permissions
- Try to create a role with the same name as another one already created
- You're going to see the error toast message

### Notes for QA
We replaced all of the methods to manipulate the roles. Would be nice to include some tests like:
- Create a new role 
- Update a role
- Delete a role
- Check if it's possible to check the users in the role

## Further comments
